### DISCUSSION TITLE
Min or max applied to buffer

### DISCUSSION THREAD

**Discussion by peteDDD** [rlogiacco/CircularBuffer#71] [General]
The operators min and max are easily applied to arrays.  Is there a straightforward way to treat the buffer as a normal array to apply these sorts of operators?

Of course, I can write my own functions to walk the buffer and provide the same results but wondering if there is a more “elegant” approach.

Thanks!

**Comment by rlogiacco** [comment]
There is a feature request to be able to copy the buffer content into an array: if that gets through and if you can accept the additional memory consumption, you can convert than compute. Otherwise the alternative is to walk through, as you were suggesting as workaround.

Any idea how we can provide an extensible friendly class which can provide any additional operation the users might need? Something along the line of a streaming operations collector class....

**Comment by peteDDD** [comment]
Thanks for the reply.  I have not studied the underlying code.  My thinking
is that if the buffer is using std::vector and just managing the first and
last iterators then applying operations like std::min_element would work.
Again, I need to dig into the code.
